Booklore crashes on Synology NAS due to AccessDeniedException in forced hidden system folders (@eaDir, #recycle, #snapshot)

When running Booklore in Docker on a Synology NAS, the application fails to start if the monitored library path contains Synology system folders such as @eaDir, #recycle, or #snapshot. These folders are automatically created by DSM for indexing, recycle bin, and snapshot features, and they are not world‑readable.
Booklore’s monitoring service attempts to walk into these directories during initialization, encounters a java.nio.file.AccessDeniedException, and aborts the entire application context. This leaves the library at 0 books and upon trying to restart the container anytime after initial setup of the library folder, it prevents the container from starting anymore.

Steps to reproduce:
- Run Booklore in Docker on a Synology NAS.
- Mount a shared folder (e.g. /volume1/eBooks) into the container.
- Ensure the folder contains Synology system directories (@eaDir, #recycle, #snapshot).
- Start the container.
- Observe that Booklore fails with `AccessDeniedException`.

Expected behavior:
Booklore should skip unreadable directories and continue scanning the rest of the library, similar to how Plex, Jellyfin, Audiobookshelf, Emby, or Calibre behave.

Actual behavior:
Booklore throws an exception during initialization and the application fails to start. This appears to be due to using `Files.walk()` without catching `AccessDeniedException`

Logs (excerpts):
```text
Caused by: java.nio.file.AccessDeniedException: /ebooks/@eaDir/SYNO@.fileindexdb
	at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source) ~[na:na]
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[na:na]
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[na:na]
	at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(Unknown Source) ~[na:na]
	at java.base/java.nio.file.Files.newDirectoryStream(Unknown Source) ~[na:na]
	at java.base/java.nio.file.FileTreeWalker.visit(Unknown Source) ~[na:na]
	at java.base/java.nio.file.FileTreeWalker.next(Unknown Source) ~[na:na]
	... 67 common frames omitted
```

```text
Caused by: java.nio.file.AccessDeniedException: /ebooks/#recycle
    at java.base/sun.nio.fs.UnixException.translateToIOException
    ...
    at com.adityachandel.booklore.service.monitoring.MonitoringService.registerLibrary
```
Environment:
- Synology DSM version: DSM 7.2.2-72806 Update 4
- Docker Compose stack (relevant volume mounts):

```TEXT
    volumes:
      - /volume1/docker/booklore/data:/app/data:rw
      - /volume1/eBooks:/ebooks:rw
      - /volume1/Audiobooks:/audiobooks:rw
      - /volume1/docker/booklore/bookdrop:/bookdrop:rw #Files inside the bookdrop folder are automatically detected.
    tmpfs:
      - /ebooks/@eaDir
      - /audiobooks/@eaDir
```
Note: Adding the tmpfs for @eaDir fixed the failure with that folder, but doing the same for #recycle did not work.

- Booklore image version: ghcr.io/booklore-app/booklore:latest

### Proposed solution:
- Wrap the directory scanning logic in a try/catch for AccessDeniedException (and possibly IOException) so that unreadable system folders are skipped instead of aborting initialization.
- Optionally, allow users to configure ignore patterns (e.g., .bookloreignore) that are applied _before_ directory registration. 

Why this matters:
- Synology users cannot avoid @eaDir, #recycle, and #snapshot — they are created automatically by DSM.
- Without this fix, Booklore is unusable on Synology for many users unless users hack around it with bindfs or tmpfs masking, or in some cases, moving all their files to a clean subdirectory instead of a root directory that they may already have established for eBooks.
- Adding skip‑on‑error behavior would make Booklore more robust across NAS environments, especially since people may intentionally restrict permissions to certain folders/files or different systems like Synology may force this as is the case here.

Other media servers (Audiobookshelf, Plex, Jellyfin) skip Synology’s system folders without crashing. Booklore should do the same. 

If anyone knows how to add this fix into the next update, please prioritize this since it makes Booklore virtually unusable and it makes sense to standardize it to function similarly to most other media servers which ignore/skip system folders with the access denied error.

--- a/booklore-api/src/main/java/com/adityachandel/booklore/service/library/LibraryFileHelper.java
+++ b/booklore-api/src/main/java/com/adityachandel/booklore/service/library/LibraryFileHelper.java
@@ -33,7 +33,9 @@ public class LibraryFileHelper {
         boolean supportsSupplementaryFiles = processor.supportsSupplementaryFiles();
 
         try (Stream<Path> stream = Files.walk(libraryPath, FileVisitOption.FOLLOW_LINKS)) {
-            return stream.filter(Files::isRegularFile)
+            return stream
+                    .filter(Files::isReadable)
+                    .filter(Files::isRegularFile)
                     .filter(path -> !FileUtils.shouldIgnore(path))
                     .map(fullPath -> {
                         String fileName = fullPath.getFileName().toString();

--- a/booklore-api/src/main/java/com/adityachandel/booklore/util/FileUtils.java
+++ b/booklore-api/src/main/java/com/adityachandel/booklore/util/FileUtils.java
@@ -8,8 +8,11 @@ import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Arrays;
 import java.util.Comparator;
+import java.util.List;
 import java.util.Optional;
+import com.adityachandel.booklore.model.entity.BookEntity;
 
 @UtilityClass
 @Slf4j
@@ -60,12 +63,19 @@ public class FileUtils {
         }
     }
 
+    private List<String> systemDirs = Arrays.asList(
+      // synology
+      "#recycle",
+      "@eaDir",
+      // calibre
+      ".caltrash"
+    );
     public boolean shouldIgnore(Path path) {
         if (!path.getFileName().toString().isEmpty() && path.getFileName().toString().charAt(0) == '.') {
             return true;
         }
         for (Path part : path) {
-            if (".caltrash".equals(part.toString())) {
+            if (systemDirs.contains(part.toString())) {
                 return true;
             }
         }

